examples/multi_gpu/papers100m_gcn_cugraph.py

```diff
@@ -0,0 +1,330 @@
+                         **kwargs)
+
+    dist.barrier()
+
+    eval_steps = 1000
+    warmup_steps = 20
+    acc = Accuracy(task="multiclass", num_classes=num_classes).to(rank)
+    dist.barrier()
+    torch.cuda.synchronize()
+    if rank == 0:
+        prep_time = round(time.perf_counter() - wall_clock_start, 2)
+        print("Total time before training begins (prep_time) =", prep_time,
+              "seconds")
+        print("Beginning training...")
+    for epoch in range(epochs):
+        train_path = os.path.join(tempdir, f'samples_{epoch}')
+
+        input_files = np.array_split(np.array(os.listdir(train_path)),
+                                     world_size)[rank]
+
+        train_loader = BulkSampleLoader(cugraph_store, cugraph_store,
+                                        directory=train_path,
+                                        input_files=input_files)
+        with Join([model], divide_by_initial_world_size=False):
+            for i, batch in enumerate(train_loader):
+                if i == warmup_steps:
+                    torch.cuda.synchronize()
+                    start = time.time()
+                batch = batch.to(rank)
+
+                if isinstance(batch, torch_geometric.data.HeteroData):
+                    batch = batch.to_homogeneous()
+                batch_size = batch.num_sampled_nodes[0]
+
+                batch.y = batch.y.to(torch.long)
+                optimizer.zero_grad()
+                out = model(batch.x, batch.edge_index)
+                loss = F.cross_entropy(out[:batch_size], batch.y[:batch_size])
+                loss.backward()
+                optimizer.step()
+                if rank == 0 and i % 10 == 0:
+                    print("Epoch: " + str(epoch) + ", Iteration: " + str(i) +
+                          ", Loss: " + str(loss))
+        nb = i + 1.0
+        dist.barrier()
+        torch.cuda.synchronize()
+        if rank == 0:
+            print("Average Training Iteration Time:",
+                  (time.time() - start) / (nb - warmup_steps), "s/iter")
+        eval_path = os.path.join(tempdir, f'samples_eval_{epoch}')
+
+        input_files = np.array(os.listdir(eval_path))
+
+        eval_loader = BulkSampleLoader(cugraph_store, cugraph_store,
+                                       directory=eval_path,
+                                       input_files=input_files)
+        with Join([model], divide_by_initial_world_size=False):
+            with torch.no_grad():
+                for i, batch in enumerate(eval_loader):
+                    if i >= eval_steps:
+                        break
+
+                    batch = batch.to(rank)
+                    if isinstance(batch, torch_geometric.data.HeteroData):
+                        batch = batch.to_homogeneous()
+                    batch_size = batch.num_sampled_nodes[0]
+
+                    batch.y = batch.y.to(torch.long)
+                    out = model.module(batch.x, batch.edge_index)
```

Comment:
Why do we need to call `module`?

Reply:
from cugraph team:
"This is because we have a different number of inputs on each rank. Without it the script will hang forever."
this will be fixed in future release of cugraph.
this is the known issue ticket for cugraph: https://github.com/rapidsai/cugraph/issues/4201

once that is fixed, i can follow up in the future and clean up this part with a quick PR